src/components/CardCart/CardCart.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useRef, useState } from 'react';
+import QuantitySelector from '../../components/QuantitySelector/QuantitySelector';
+import { useCounter } from '../../utils/hooks/useCounter';
+import styles from './CardCart.module.css';
+import { useGlobalContext } from '../../context/Provider';
+
+function CardCart({ id, name, price, image, imageAlt, stock, quantity }) {
+    const { addItemToCart, deleteItemFromCart, itemsInCart } = useGlobalContext();
+    const counter = useCounter(quantity);
+    const firstUpdate = useRef(true);
+    const [isDisabled, setIsDisabled] = useState(false);
+
+    useEffect(() => {
+        counter.value >= stock ? setIsDisabled(true) : setIsDisabled(false);
```

Comment:
`counter.value >= stock` is a boolean expression, it can be used instead of the ternary:

```suggestion
        setIsDisabled(counter.value >= stock);
```